### PR TITLE
INGK-520 Update save_configuration function

### PR DESCRIPTION
**Pull request by pzinemanas**
This PR updates the `save_configuration` function so the xcf file is created based on the registers stored at the dictionary. It also improves the test related to this function so not only checks if the file exists.

### CODE REVIEW

File: tests/servo/test_servo.py [hunk 1]
```diff
@@ -1,18 +1,68 @@
...
+        if subnode not in subnodes:
+            continue
+
+        reg_id = saved_register.attrib.get('id')
+        registers = servo.dictionary.registers(subnode=subnode)
+
+        assert reg_id in registers
+
+        storage = saved_register.attrib.get('storage')
+        if storage is not None:
```
Comment (MartinALL91) [inline tests/servo/test_servo.py:54]: With Python 3.8+ you could do:

```
reg_storage = registers[reg_id].storage
if (saved_storage := saved_register.attrib.get('storage')) is not None:
    reg_storage = str(reg_storage)
assert reg_storage == saved_storage

**Review by MartinALL91 — APPROVED** [review]

File: tests/servo/test_servo.py (hunk 1 above)
Comment (pzinemanas) [reply tests/servo/test_servo.py:50]: Interesting, but the library supports py36 and 38 as well, right?

Comment (MartinALL91) [reply tests/servo/test_servo.py:50]: Yes, the comment was just to show how we could simplify this part of the code by using the walrus operator.

**Comment by GullinBustin** [comment]
As you say in the comment, remove  `servo.__update_single_axis_dict` and `servo.__update_multiaxis_dict` functions

**Review by GullinBustin — CHANGES_REQUESTED** [review]
As you say in the comment, remove `servo.__update_single_axis_dict` and `servo.__update_multiaxis_dict` functions

**Review by GullinBustin — APPROVED** [review]

**Pull request merged**